app/components/additional-fields-input.js

```diff
@@ -0,0 +1,63 @@
+import Ember from "ember";
+import config from "../config/environment";
+import { pluralize } from "ember-inflector";
+import _ from "lodash";
+
+export default Ember.TextField.extend({
+  tagName: "input",
+  type: "text",
+  isMobileApp: config.cordova.enabled,
+  subformDetailService: Ember.inject.service(),
+  attributeBindings: [
+    "name",
+    "type",
+    "value",
+    "maxlength",
+    "id",
+    "autoFocus",
+    "placeholder",
+    "required",
+    "pattern"
+  ],
+  store: Ember.inject.service(),
+  classNameBindings: ["inlineTextInput"],
+  inlineTextInput: false,
+  previousValue: "",
+
+  focusOut() {
+    const detailType = _.snakeCase(this.get("detailType")).toLowerCase();
+    const apiEndpoint = pluralize(detailType);
+    const detailId = this.get("detailId");
+    const url = `/${apiEndpoint}/${detailId}`;
```

Comment:
We want to avoid any **business logic/api** code within components as much as possible, because it makes them **unreusable** them in other pages/contexts.

They should be as **generic** as possible, an input field is just an input field (should not have hidden magic).
I recommend passing a callback from the controller to the component (e.g `onFocusOut=(action 'saveField' fieldName)`), so the logic lives outside of it.

So far other components have not followed those practices, which in turn forces us to copy them into many practically identical components. 

However:
There are some cases for some business logic components, usually big chunks of a page which are complex and not meant to be customised in any way
If you are in that situation, notice that I created a `components/goodcity/` folder within. 
That's what that is for, we prefix it by 'goodcity' to separate it from the others which are simply our own UI toolkit.
You can place a generic reusable component under `components/` and a custom one that extends it under `components/goodcity`


Apologies for the rant 😄